tools/download_test.go

```diff
@@ -25,3 +29,67 @@ func Test_findBaseDir(t *testing.T) {
 		})
 	}
 }
+
+func TestTools_DownloadAndUnpackBehaviour(t *testing.T) {
+	urls := []string{
+		"http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino14-armhf-pc-linux-gnu.tar.bz2",
+		"http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino14-aarch64-pc-linux-gnu.tar.bz2",
+		"http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino14-i386-apple-darwin11.tar.bz2",
+		"http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino14-x86_64-pc-linux-gnu.tar.bz2",
+		"http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino14-i686-pc-linux-gnu.tar.bz2",
+		"http://downloads.arduino.cc/tools/avrdude-6.3.0-arduino14-i686-w64-mingw32.zip",
+	}
+	expectedDirList := []string{"bin", "etc"}
+
+	for _, url := range urls {
+		t.Log("Downloading tool from " + url)
+		resp, err := http.Get(url)
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+		defer resp.Body.Close()
+
+		// Read the body
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+
+		location := path.Join("/tmp", dir(), "arduino", "avrdude", "6.3.0-arduino14")
+		os.MkdirAll(location, os.ModePerm)
+		err = os.RemoveAll(location)
+
+		if err != nil {
+			t.Errorf("%v", err)
+		}
+
+		srcType, err := mimeType(body)
```

Comment:
Don't we want to test the internal function that takes care of format/mime type here?